route/route.go

```diff
@@ -860,6 +845,21 @@ func (r *Router) readAndCloseMaybeCompressedBody(req *http.Request) (*bytes.Buff
 		return nil, err
 	}
 
+	if encodingScheme == "zstd" {
```

Comment:
please add a comment here to give more insight to future us as to why this is outside of the switch now